### PR TITLE
Docker mvp

### PR DESCRIPTION
**Pull request by gabelerner**

### CODE REVIEW

File: server/controllers/api/v1/compounds.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ var compoundToJson = (compound) => {
 
 module.exports.show = function(req, res) {
   CompoundView.findOne({
-    cid: req.params.cid
+    where: { cid: req.params.cid }
```
Comment (gabelerner) [inline server/controllers/api/v1/compounds.js]: bug

File: server/controllers/api/v1/docking.js [hunk 2]
```diff
@@ -1,5 +1,35 @@
+const uuid = require('uuid');
+const fetch = require('node-fetch');
+const queue = require('../../../queue');
+const bin = require('../../../util/bin');
+const { CompoundView } = require('../../../models/compoundView');
+const results = require('../../../worker/results');
+
 module.exports.start = function(req, res) {
+  const structure = req.params.structure;
+  const url = `https://files.rcsb.org/download/${structure.toUpperCase()}.pdb`;
```
Comment (gabelerner) [inline server/controllers/api/v1/docking.js]: TODO: move below to job

File: server/util/bin.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const os = require('os');
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+const cp = require('child_process');
```
Comment (gabelerner) [inline server/util/bin.js]: filesystem/binary helper methods

File: server/worker/index.js [hunk 4]
```diff
@@ -1,13 +1,39 @@
...
-  done(null, {
-    foo: 'bar',
-    ligandCid: job.data.ligandCid,
-    pdbLink: job.data.pdbLink
+  console.log(`processing job ${job.data.jobId} (cid: ${job.data.ligandCid})`);
+
+  CompoundView.findOne({
+    where: { cid: job.data.ligandCid },
+  }).then(ligand => {
+    var dir = bin.tmpDir();
```
Comment (gabelerner) [inline server/worker/index.js]: logic for a job is here

File: server/worker/results.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+var redis = require("redis"),
+    client = redis.createClient();
+
+client.on("error", function (err) {
+    console.log("Error " + err);
+});
+
+module.exports.addResult = function(jobId, ligandCid, score) {
```
Comment (gabelerner) [inline server/worker/results.js]: using redis sorted sets to add and then return job data

**Comment by jsdir** [comment]
looks good

**Pull request merged**